File: lc_go/lcGo/prefix/canEat.go
package main

func canEat(candiesCount []int, queries [][]int) []bool {
	n := len(candiesCount)

	prefix := make([]int, n+1)

	for i := 0; i < n; i++ {
		prefix[i+1] = prefix[i] + candiesCount[i]
	}
	ans := make([]bool, len(queries))

	for i := 0; i < len(queries); i++ {
		CandyType := queries[i][0]
		favoriteDay := queries[i][1]
		dailyCapacity := queries[i][2]
		minTaking := favoriteDay + 1
		maxTaking := minTaking * dailyCapacity
		currCandy := prefix[CandyType] + 1
		nxtCandy := prefix[CandyType+1]
		ans[i] = !(maxTaking < currCandy || minTaking > nxtCandy)
	}
	return ans
}

func main() {

}

File: lc_py/lcPy/prefix/canEat.py
from typing import List


class Solution:
    def canEat(self, candiesCount: List[int], queries: List[List[int]]):
        ans = [False] * len(queries)

        prefix = [0] * (len(candiesCount) + 1)

        for i, x in enumerate(candiesCount):
            prefix[i + 1] = prefix[i] + x

        for i in range(len(queries)):
            min_taking = queries[i][1] + 1
            max_taking = min_taking * queries[i][2]

            curr_candy = prefix[queries[i][0]] + 1
            nxt_candy = prefix[queries[i][0] + 1]

            ans[i] = not (min_taking > nxt_candy or max_taking < curr_candy)
        return ans
